cellrank/kernels/_monge_kernel.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Tuple, Union, Literal, Callable, Optional, Sequence
+
+from moscot.problems.time import TemporalNeuralProblem
+
+from cellrank._utils._docs import d, inject_docs
+from scvelo.preprocessing.moments import get_moments
+from cellrank.kernels._velocity_kernel import VelocityKernel
+from cellrank.kernels.utils._similarity import Similarity
+from cellrank.kernels.utils._velocity_model import BackwardMode, VelocityModel
+
+import numpy as np
+
+__all__ = ["MongeKernel"]
+
+_ATTR_OPTIONS = Literal["obsm", "layers"]
+
+
+@d.dedent
+class MongeKernel(VelocityKernel):
+    """
+    Summary.
+
+    Description.
+
+    Args:
+        VelocityKernel (_type_): _description_
+    """
+
+    def __init__(
+        self,
+        tnp: TemporalNeuralProblem,
```

Comment:
I don't think we need to pass `TemporalNeuralProblem`, as you just extract the `adata` attribute (not sure whether it has `{x,v}keys` and `attr` you could extract from it, but if yes, in that case it would make more sense.
Which leaves the question - why is this problem class even needed, apart from setting certain defaults in `__init__`? Am fine with having a separate class, as long as, e.g., `compute_transition_matrix` does something "useful", like restricts which velocity modes are available (line right now).
Otherwise, would just refactor `_read_from_adata` to accept how velocity data can be stored (obsm) instead of defining this kernel, which would seems a bit redundant.

Reply:
I am not sure I understand this sentence: "as you just extract the adata attribute (not sure whether it has {x,v}keys and attr you could extract from it, but if yes, in that case it would make more sense."

I think we really only need to take the `adata` that is stored in the `TemporalNeuralProblem` and `xkey`, `vkey` and `attr` only refer to the extracted `adata`, no other info in `TemporalNeuralProblem`.

Reply:
After reading your comment I realize that I previously also misunderstood the three different `model` options in the `VelocityKernel`'s `compute_transition_matrix()`: I thought these refer to different `scVelo` velocity models (deterministic, stochastic, dynamic or something like that) but they are really `cellrank` specific models of how to compute the transition matrix, right? In that case it doesn't actually make sense to hard-code to the deterministic model as I (currently implemented it) but to keep all options.